dataprofiler/labelers/base_model.py

```diff
@@ -74,6 +74,7 @@ def __eq__(self, other: BaseModel) -> bool:  # type: ignore
         """
         if (
             type(self) != type(other)
+            or not isinstance(other, BaseModel)
```

Comment:
This theoretically should be covered by above, but does mypy get confused?

Reply:
yeah mypy thinks `other` is still an object

Reply:
`type(self) != type(other)` should validate that `not isinstance(other, BaseDataProcessor)` would return `True`. Just a redundant check. I'm fine with it --> better than `#type: ignore` and casting IMO